frontend/src/demo.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-console */
+/**
+ * Type checking test file for posthog-typed.ts
+ *
+ * Verifies that:
+ * 1. capture() validates required properties for defined events
+ * 2. capture() allows additional properties beyond the schema
+ * 3. capture() accepts undefined events with any properties
+ * 4. capture() blocks broad string type variables
+ * 5. captureRaw() bypasses all type checking
+ */
+import posthog, { EventName } from './lib/posthog-typed'
+
+// ========================================
+// Defined events: Type safety with flexibility
+// ========================================
+
+// ✅ Required property only
+posthog.capture('downloaded_file', {
+    file_size_b: 1024000,
+})
+
+// ✅ Additional properties allowed (key feature!)
```

Comment:
Haha I can definitely imagine people asking us to disable this feature. Happy to wait for this request to happen before we deal with it though